internal/docstore/driver/driver.go

```diff
@@ -24,6 +24,12 @@ import (
 
 // A Collection is a set of documents.
 type Collection interface {
+	// Key returns the document key, or nil if the document doesn't have one. Key
+	// should not return an error if the key is missing unless the collection cannot
```

Comment:
The `unless the collection ...` part reads a little bit hard, maybe separate into its own sentence and explain what it means by `cannot generate a fresh key`?

Reply:
Done.